### PR TITLE
Feature/ae 367 tasmaytysraportti

### PR DESCRIPTION
**Pull request by MattiNieminen**

### CODE REVIEW

File: etp-backend/src/main/clj/solita/etp/service/laskutus.clj [hunk 1]
```diff
@@ -377,9 +391,8 @@
                                        laskutustieto-xml-files
                                        laskutustieto-destination-dir)
               (log/info "Laskutustieto xmls uploaded with SFTP."))
-
-            ;; TODO send täsmätytysraportti with SMTP.
-            )
+            (send-tasmaytysraportti-email! tasmaytysraportti-file)
+            (log/info "Täsmätysraportti sent as an email"))
         (log/warn "SFTP configuration missing. Skipping actual integration."))
       (delete-files! all-files)
```
Comment (muep) [inline etp-backend/src/main/clj/solita/etp/service/laskutus.clj:397]: It is not changed in this change set, but I wonder if this delete should be in the `finally` section.

File: etp-backend/src/main/clj/solita/etp/service/laskutus.clj [hunk 2]
```diff
@@ -377,9 +391,8 @@
                                        laskutustieto-xml-files
                                        laskutustieto-destination-dir)
               (log/info "Laskutustieto xmls uploaded with SFTP."))
-
-            ;; TODO send täsmätytysraportti with SMTP.
-            )
+            (send-tasmaytysraportti-email! tasmaytysraportti-file)
+            (log/info "Täsmätysraportti sent as an email"))
```
Comment (muep) [inline etp-backend/src/main/clj/solita/etp/service/laskutus.clj:395]: It's just a nice to have, but might be nice to one or both of the destination address and the message ID that SMTP servers give us when they accept messages in.

File: etp-backend/src/main/clj/solita/etp/service/laskutus.clj [hunk 3]
```diff
@@ -293,6 +294,19 @@
                                :xlsx (.getName xlsx-file)
                                :pdf-result? pdf-exists?)))))))
 
+(defn send-tasmaytysraportti-email! [tasmaytysraportti-file]
+  (smtp/send-email! config/smtp-host
```
Comment (muep) [inline etp-backend/src/main/clj/solita/etp/service/laskutus.clj:298]: packing these SMTP parameters to a map or a record might help a bit with readability.

**Review by muep — APPROVED** [review]

File: etp-backend/src/main/clj/solita/etp/service/laskutus.clj (hunk 3 above)
Comment (MattiNieminen) [reply etp-backend/src/main/clj/solita/etp/service/laskutus.clj:298]: Thought about that too. But then realized that there are probably use cases where someone wants to another function (with partial) where host, port, etc are already bound. That's not possible (with partial at least) if arguments are inside a map.

File: etp-backend/src/main/clj/solita/etp/service/laskutus.clj (hunk 1 above)
Comment (MattiNieminen) [reply etp-backend/src/main/clj/solita/etp/service/laskutus.clj:397]: That's true. Would require some refactoring at some point as filenames are created inside try-block, so deleting them is not possible using the filenames.

**Pull request merged**